cmd/podman/pods/create.go

```diff
@@ -144,6 +148,10 @@ func create(cmd *cobra.Command, args []string) error {
 			return errors.New("cannot set infra-image without an infra container")
 		}
 		createOptions.InfraImage = ""
+		if createOptions.InfraName != "" {
+			return errors.New("cannot set infra-name without an infra container")
+		}
+		createOptions.InfraName = ""
```

Comment:
I'm confused. You just tested against empty string. The only way the code can get here is if `InfraName` is already empty string. What purpose does this line serve?